app/build.gradle

```diff
@@ -117,78 +117,51 @@ dependencies {
     implementation 'org.briarproject:moat-api:0.2'
     implementation 'com.squareup.okhttp3:okhttp:4.10.0'
 
-    implementation 'info.guardianproject:tor-android:0.4.6.10'
+    implementation 'info.guardianproject:tor-android:0.4.7.8'
     implementation 'info.guardianproject:jtorctl:0.4.5.7'
-    tor 'org.briarproject:obfs4proxy-android:0.0.14'
-    tor 'org.briarproject:snowflake-android:2.3.1'
+    tor 'org.briarproject:obfs4proxy-android:0.0.14-tor2'
+    tor 'org.briarproject:snowflake-android:2.5.1'
 
-    def ktor_version = '2.2.2'
+    def ktor_version = '2.2.3'
```

Comment:
While testing this branch I got the following crash on API 21:

```
13:42:34.891  E  FATAL EXCEPTION: DefaultDispatcher-worker-8
                 Process: org.onionshare.android.nightly, PID: 7851
                 java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/function/Function;
                 	at com.mitchellbosecke.pebble.extension.ExtensionRegistryFactory.<init>(ExtensionRegistryFactory.java:26)
                 	at com.mitchellbosecke.pebble.PebbleEngine$Builder.<init>(PebbleEngine.java:301)
                 	at io.ktor.server.pebble.PebbleConfiguration.<init>(Pebble.kt:20)
                 	at io.ktor.server.pebble.PebbleKt$Pebble$1.invoke(Pebble.kt:51)
                 	at io.ktor.server.pebble.PebbleKt$Pebble$1.invoke(Pebble.kt:51)
                 	at io.ktor.server.application.CreatePluginUtilsKt.createPluginInstance(CreatePluginUtils.kt:261)
                 	at io.ktor.server.application.CreatePluginUtilsKt.access$createPluginInstance(CreatePluginUtils.kt:1)
                 	at io.ktor.server.application.CreatePluginUtilsKt$createApplicationPlugin$2.install(CreatePluginUtils.kt:90)
                 	at io.ktor.server.application.CreatePluginUtilsKt$createApplicationPlugin$2.install(CreatePluginUtils.kt:83)
                 	at io.ktor.server.application.ApplicationPluginKt.install(ApplicationPlugin.kt:98)
                 	at org.onionshare.android.server.WebserverManager$start$2.invoke(WebserverManager.kt:69)
                 	at org.onionshare.android.server.WebserverManager$start$2.invoke(WebserverManager.kt:64)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:320)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading$instantiateAndConfigureApplication$1.invoke(ApplicationEngineEnvironmentReloading.kt:309)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.avoidingDoubleStartup(ApplicationEngineEnvironmentReloading.kt:337)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.instantiateAndConfigureApplication(ApplicationEngineEnvironmentReloading.kt:309)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.createApplication(ApplicationEngineEnvironmentReloading.kt:150)
                 	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.start(ApplicationEngineEnvironmentReloading.kt:276)
                 	at io.ktor.server.netty.NettyApplicationEngine.start(NettyApplicationEngine.kt:212)
                 	at io.ktor.server.netty.NettyApplicationEngine.start(NettyApplicationEngine.kt:30)
                 	at io.ktor.server.engine.ApplicationEngine$DefaultImpls.start$default(ApplicationEngine.kt:68)
                 	at org.onionshare.android.server.WebserverManager.start(WebserverManager.kt:78)
                 	at org.onionshare.android.ShareManager$shareState$1.invokeSuspend(ShareManager.kt:82)
                 	at org.onionshare.android.ShareManager$shareState$1.invoke(ShareManager.kt)
                 	at org.onionshare.android.ShareManager$shareState$1.invoke(ShareManager.kt)
                 	at kotlinx.coroutines.flow.FlowKt__ZipKt$combineTransform$$inlined$combineTransformUnsafe$FlowKt__ZipKt$4$1.invokeSuspend(Zip.kt:333)
                 	at kotlinx.coroutines.flow.FlowKt__ZipKt$combineTransform$$inlined$combineTransformUnsafe$FlowKt__ZipKt$4$1.invoke(Zip.kt)
                 	at kotlinx.coroutines.flow.FlowKt__ZipKt$combineTransform$$inlined$combineTransformUnsafe$FlowKt__ZipKt$4$1.invoke(Zip.kt)
                 	at kotlinx.coroutines.flow.internal.CombineKt$combineInternal$2.invokeSuspend(Combine.kt:79)
                 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
                 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
                 	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@8149e24, Dispatchers.Default]
                 Caused by: java.lang.ClassNotFoundException: Didn't find class "java.util.function.Function" on path: DexPathList[[zip file "/data/app/org.onionshare.android.nightly-1/base.apk"],nativeLibraryDirectories=[/data/app/org.onionshare.android.nightly-1/lib
```

Enabling library desugaring solves this (tested with desugar_jdk_libs 2.0.2).

At first I thought this was due to upgrading Pebble, but it also happens on the main branch.

Reply:
I've added a commit that enables library desugaring.

Reply:
This may also solve https://github.com/onionshare/onionshare-android/issues/80